src/service/CinemaService.java

```diff
@@ -0,0 +1,179 @@
+package service;
+
+import exception.AlreadyTakenSeatException;
+import exception.NonExistingSeatException;
+
+import java.util.Arrays;
+import java.util.Scanner;
+
+public class CinemaService {
+
+    private final Cinema cinema;
+
+    public CinemaService() {
+        cinema = initializeCinema();
+        createSeats();
+    }
+
+    private Cinema initializeCinema() {
+        Scanner scanner = new Scanner(System.in);
```

Comment:
Can you create a single instance of Scanner for CinemaService?

Reply:
Extracted Scanner initialization and creation from methods to field/constructor